src/mariadb.ts

```diff
@@ -0,0 +1,68 @@
+const mariadb = require("mariadb");
+const uuidv4 = require('uuid/v4');
+
+export class MariaDB {
+    constructor() {
+
+    }
+
+    public async register(username: string, password: string) {
+        const conn = await this.createConnection()
+        const otherUsers = await conn.query(
+            `SELECT * 
+             FROM Users
+             WHERE UserName="${username}"`
+        );
+        if (otherUsers.length > 0) {
+            conn.end();
+            throw new Error('Username is taken');
+        }
+        await conn.query(
+		    `INSERT INTO Users (UserName, Password, Credit) values (
+			    "${username}",
+                "${password}",
+                0
+            )`);
+        conn.end();
+        
+        return await this.loginUser(username, password);
+    }
+
+    public async loginUser(username: string, password: string) {
+        const conn = await this.createConnection();
+        const requestedUser = await conn.query(
+            `SELECT *
+             FROM Users
+             WHERE UserName="${username}"`
+        );
+        if (requestedUser.length == 0) {
+            throw new Error('Invalid Username');
+        }
+        if (requestedUser[0].Password != password) {
+            throw new Error('Invalid Password')
+        }
+        const sessionToken = uuidv4().substring(0, 16);
+        const expirationDate = new Date();
+        expirationDate.setTime(expirationDate.getTime() + (24 * 60 * 60 * 1000));
+        await conn.query(
+            `INSERT INTO LoginSessions (SessionToken, UserID, Expires) values (
+                "${sessionToken}",
+                "${requestedUser[0].UserID}",
+                "${expirationDate.toISOString().slice(0, 19).replace('T', ' ')}")
+            `
+        );
+        conn.end();
+        return sessionToken;
+    }
+
+    private createConnection() {
+        return  mariadb.createConnection({
+            // Open a new connection
+            user: "db_admin",
+            database: "story_exchange",
+            host: "localhost",
+            password: "fake42",
```

Comment:
Just as a note, but I don't think it matters for our use case since its just for class, but I think with stuff like this, you would usually set the password as an environment variable or put it in a config file that you don't push to github, and then access it from there. But we don't need to worry about that, just figured I'd point it out.

Reply:
Yes, this is definitely a bad way to do it. I felt like it was acceptable for our situation though.